auth/src/main/java/com/firebase/ui/auth/data/client/CountryListLoadTask.java

```diff
@@ -38,267 +42,52 @@
-        countryInfoList.add(new CountryInfo(new Locale("", "TJ"), 992));
-        countryInfoList.add(new CountryInfo(new Locale("", "TZ"), 255));
-        countryInfoList.add(new CountryInfo(new Locale("", "TH"), 66));
-        countryInfoList.add(new CountryInfo(new Locale("", "TG"), 228));
-        countryInfoList.add(new CountryInfo(new Locale("", "TK"), 690));
-        countryInfoList.add(new CountryInfo(new Locale("", "TO"), 676));
-        countryInfoList.add(new CountryInfo(new Locale("", "TT"), 1));
-        countryInfoList.add(new CountryInfo(new Locale("", "TN"), 216));
-        countryInfoList.add(new CountryInfo(new Locale("", "TR"), 90));
-        countryInfoList.add(new CountryInfo(new Locale("", "TM"), 993));
-        countryInfoList.add(new CountryInfo(new Locale("", "TC"), 1));
-        countryInfoList.add(new CountryInfo(new Locale("", "TV"), 688));
-        countryInfoList.add(new CountryInfo(new Locale("", "VI"), 1));
-        countryInfoList.add(new CountryInfo(new Locale("", "UG"), 256));
-        countryInfoList.add(new CountryInfo(new Locale("", "UA"), 380));
-        countryInfoList.add(new CountryInfo(new Locale("", "AE"), 971));
-        countryInfoList.add(new CountryInfo(new Locale("", "GB"), 44));
-        countryInfoList.add(new CountryInfo(new Locale("", "US"), 1));
-        countryInfoList.add(new CountryInfo(new Locale("", "UY"), 598));
-        countryInfoList.add(new CountryInfo(new Locale("", "UZ"), 998));
-        countryInfoList.add(new CountryInfo(new Locale("", "VU"), 678));
-        countryInfoList.add(new CountryInfo(new Locale("", "VA"), 379));
-        countryInfoList.add(new CountryInfo(new Locale("", "VE"), 58));
-        countryInfoList.add(new CountryInfo(new Locale("", "VN"), 84));
-        countryInfoList.add(new CountryInfo(new Locale("", "WF"), 681));
-        countryInfoList.add(new CountryInfo(new Locale("", "EH"), 212));
-        countryInfoList.add(new CountryInfo(new Locale("", "YE"), 967));
-        countryInfoList.add(new CountryInfo(new Locale("", "ZM"), 260));
-        countryInfoList.add(new CountryInfo(new Locale("", "ZW"), 263));
+        List<CountryInfo> countryInfoList = getAvailableCountryCodes();
         Collections.sort(countryInfoList);
         return countryInfoList;
     }
 
+    public List<CountryInfo> getAvailableCountryCodes() {
+        Map<String, Integer> countryInfoMap = PhoneNumberUtils.getImmutableCountryCodeMap();
+        if (whitelistedCountryCodes == null && blacklistedCountryCodes == null) {
+            whitelistedCountryCodes = new ArrayList<>(countryInfoMap.keySet());
+        }
+
+        List<CountryInfo> countryInfoList = new ArrayList<>();
+
+        Set<String> excludedCountries = new HashSet<>();
+        if (whitelistedCountryCodes == null) {
+            excludedCountries.addAll(blacklistedCountryCodes);
+        } else {
+            excludedCountries.addAll(countryInfoMap.keySet());
```

Comment:
Add some comments here, as the assumptions took me a minute to wrap my head around (I think you're saying that at this point you know only one of whitelist or blacklist will be non-null so you're just handling the two cases for whitelist and assuming the opposite about blacklist)

Reply:
Done.